ui/src/components/friends/add.rs

```diff
@@ -189,44 +189,88 @@ pub fn AddFriend(cx: Scope) -> Element {
+                                }
+                                if STATIC_ARGS.use_mock {
+                                    if let Ok(did) = DID::from_str(friend_input.get()) {
+                                        let mut ident = Identity::default();
+                                        ident.set_did_key(did);
+                                        state.write().mutate(Action::SendRequest(ident));
+                                    }
+                                } else {
+                                    add_in_progress.set(true);
+                                    ch.send((friend_input.get().to_string(), state.read().outgoing_fr_identities()));
+                                }
+                            },
+                            onchange: |(s, is_valid)| {
+                                friend_input.set(s);
+                                friend_input_valid.set(is_valid);
+                            },
+                            aria_label: "Add Someone Input".into()
                         }
-                        if STATIC_ARGS.use_mock {
-                            if let Ok(did) = DID::from_str(friend_input.get()) {
-                                let mut ident = Identity::default();
-                                ident.set_did_key(did);
-                                state.write().mutate(Action::SendRequest(ident));
+                    )),
+                    items: cx.render(rsx!(
+                        ContextItem {
+                            icon: Icon::ClipboardDocument,
+                            aria_label: "friend-add-input-copy".into(),
+                            text: get_local_text("uplink.copy-text"),
+                            onpress: move |_| {
+                                let text = friend_input.get().to_string();
+                                match Clipboard::new() {
+                                    Ok(mut c) => {
+                                        if let Err(e) = c.set_text(text) {
+                                            log::warn!("Unable to set text to clipboard: {e}");
+                                        }
+                                    }
+                                    Err(e) => {
+                                        log::warn!("Unable to create clipboard reference: {e}");
+                                    }
+                                };
+                            }
+                        },
+                        ContextItem {
+                            icon: Icon::ClipboardDocument,
+                            aria_label: "friend-add-input-paste".into(),
+                            text: get_local_text("uplink.paste"),
+                            onpress: move |_| {
+                                match Clipboard::new() {
+                                    Ok(mut c) => {
+                                        let text = c.get_text().unwrap_or_default();
```

Comment:
In the event there is an error using this function, we should log so we would know in case it comes back to uses the default string due to an error.

```suggestion
                                        let text = c.get_text().map_err(|e| {
                                             log::error!("Unable to get clipboard content: {e});
                                             e
                                        }).unwrap_or_default();
```